src/qtcwin.c

```diff
@@ -0,0 +1,1816 @@
+			    strlen(qtcreclist.callsign) > 0 &&
+			    qtcreclist.serial > 0 &&
+			    qtcreclist.count > 0 &&
+			    qtcreclist.confirmed == 0
+			) {
+			    if (trxmode == CWMODE) {
+				sendmessage(qtc_recv_msgs[1]);
+			    }
+			    if (trxmode == SSBMODE) {
+				play_file(qtc_phrecv_message[1]);
+			    }
+			    if (qtcrec_record == 1 && record_run < 0) {
+				strcpy(reccommand, qtcrec_record_command[0]);
+				get_time();
+				strftime(tempc, 60, "%y%m%d%H%M%S.wav", time_ptr);
+				strcat(reccommand, tempc);
+				strcat(reccommand, qtcrec_record_command[1]);
+				record_run = system(reccommand);
+			    }
+			    activefield++;
+			    showfield(activefield);
+			}
+		    }
+		    if (trxmode == DIGIMODE) {
+			if (direction == SEND && (activefield == 0 || activefield == 2) && qtclist.totalsent == 0) {
+			    if (qtc_send_msgs[1][strlen(qtc_send_msgs[1])-1] == 10) {
+				qtc_send_msgs[1][strlen(qtc_send_msgs[1])-1] = '\0';
+			    }
+			    tlen = strlen(qtc_send_msgs[1])-5; // len("sr/nr") = 5
+			    char tmess[300], timec[40];
+			    int ql;
+			    tmess[0] = '\0';
+			    if (tlen > 0 && strncmp(qtc_send_msgs[1] + tlen, "sr/nr", 5) == 0) {
+				tempc[0] = '\0';
+				strncpy(tempc, qtc_send_msgs[1], tlen-1);
+				tempc[tlen-1] = '\0';
+				sprintf(tmess, "%s %d/%d %s %d/%d\n", tempc, qtclist.serial, *qtccount, tempc, qtclist.serial, *qtccount);
+			    }
+			    timec[0] = '\0';
+			    get_time();
+			    strftime(timec, 40, "%d-%b-%y %H:%M", time_ptr);
+
+			    for(ql = 0; ql < *qtccount; ql++) {
+				qtclist.qtclines[ql].sent = 1;
+				strncpy(qtclist.qtclines[ql].senttime, timec, 15);
+				qtclist.qtclines[ql].senttime[15] = '\0';
+				qtclist.totalsent++;
+
+				tempc[0] = '\0';
+				strip_spaces(qtclist.qtclines[ql].qtc, tempc);
+				strcat(tmess, tempc);
+				mvwprintw(qtcwin, ql+3, 30, "*");
+				qtclist.qtclines[ql].flag = 1;
+				
+			    }
+			    data_ready = 1;
+			    strncpy(wkeyerbuffer, tmess, strlen(tmess));
+			    write_keyer();
+			    wattrset(qtcwin, line_inverted);
+			    mvwprintw(qtcwin, 2, 11, "CTRL+S to SAVE!");
+			    refreshp();
+			    show_help_msg(6);
+
+			    showfield(activefield);
+			}
+			else {
+			  if (direction == SEND) {
+				tempc[0] = '\0';
+				strip_spaces(qtclist.qtclines[activefield-3].qtc, tempc);
+				sendmessage(tempc);
+			    }
+			}
+		    }
+		    break;
+	  case 19:	// CTRL-S - save QTC
+		    if (*qtccurrdiretion == SEND && *qtccount > 0 && qtclist.totalsent == *qtccount) {
+			if (log_sent_qtc_to_disk(nr_qsos) == 0) {
+			    wattrset(qtcwin, line_inverted);
+			    mvwprintw(qtcwin, 2, 11, "QTC's have been saved!");
+			    prevqtccall[0] = '\0';
+			    qtccallsign[0] = '\0';
+			    refreshp();
+			    sleep(1);
+			    x = 27;	// close the window
+			}
+			//data_ready = 1;
+		    }
+		    if (*qtccurrdiretion == RECV && trxmode == DIGIMODE) {
+			qtc_ry_capture = 1;
+			wattr_get(qtcwin, &attributes, &cpair, NULL);
+			wattrset(qtcwin, line_inverted);
```

Comment:
May be nobody needs that. I could't solve another way to store the problem. Should I do anything at this issue?